modules/persistence/src/services/metadata/associated_products/index.ts

```diff
@@ -52,54 +69,68 @@ const getAllAssociatedRepoBranchesEntries = async (metadata: SharedMetadata) =>
+      const { url, prefix } = entry;
+      const branches = Object.fromEntries(entry.branches.map((branch) => [branch.gitBranchName, { ...branch }]));
+      return [entry.project, { ...branches, url, prefix }];
     })
   );
 
-const hasAssociations = (metadata) => !!metadata.associated_products?.length;
+export const hasAssociations = (metadata) => !!metadata.associated_products?.length;
 
-// TODO: update param to target associated project name.
-// confusing - takes in associated metadata and returns umbrella metadata
-const umbrellaMetadataEntry = async (metadata): Promise<SharedMetadata> => {
+const umbrellaMetadataEntry = async (project: string): Promise<Metadata> => {
   try {
     const snooty = await db();
     const entry = await snooty
       .collection('metadata')
       .find({
-        'associated_products.name': metadata.project,
+        'associated_products.name': project,
       })
       .sort({ build_id: -1 })
       .limit(1)
       .toArray();
-    return entry[0] as unknown as SharedMetadata;
+    return entry[0] as unknown as Metadata;
   } catch (error) {
     console.log(`Error at time of querying for umbrella metadata entry: ${error}`);
     throw error;
   }
 };
 
-// Convert our cursor from the shared ToC aggregation query into a series of ToC objects
+// Convert our cursor from the associated metadata aggregation query into a series of ToC objects and their parent metadata entries
 const shapeToCsCursor = async (
   tocCursor: AggregationCursor,
   repoBranchesMap
-): Promise<{ tocInsertions: ToCInsertions; tocOrderInsertions: TocOrderInsertions }> => {
+): Promise<{
+  tocInsertions: ToCInsertions;
+  tocOrderInsertions: TocOrderInsertions;
+  associatedMetadataEntries: Metadata[];
+}> => {
   const tocInsertions = {};
   const tocOrderInsertions = {};
+  const associatedMetadataEntries: Metadata[] = [];
 
-  await tocCursor.forEach((doc) => {
+  await tocCursor.forEach((doc: AggregatedMetadata) => {
+    const { project, branch } = doc._id;
+    const metadata = doc.most_recent;
     // Initialize to empty object if we haven't already, for a given project.
-    if (!tocInsertions[doc._id.project]) tocInsertions[doc._id.project] = {};
-    if (!tocOrderInsertions[doc._id.project]) tocOrderInsertions[doc._id.project] = {};
+    if (!tocInsertions[project]) tocInsertions[project] = {};
+    if (!tocOrderInsertions[project]) tocOrderInsertions[project] = {};
 
+    const projectEntry = repoBranchesMap?.[project];
+    const branchesEntry = projectEntry?.[branch];
     // TODO: If we want staging builds with embedded versions, it needs to be added here
-    if (repoBranchesMap?.[doc._id.project]?.[doc._id.branch]) {
-      tocInsertions[doc._id.project][doc._id.branch] = doc.most_recent.toctree;
-      tocOrderInsertions[doc._id.project][doc._id.branch] = doc.most_recent.toctreeOrder;
+    if (branchesEntry) {
+      const { url, prefix } = prefixFromEnvironment(branchesEntry);
```

Comment:
minor. projectEntry here

Reply:
I think var names just need to be improved here - we have [project][branch], which is confusing since we've discussed `repoBranches` really being a "project" entry


but the [project][branch] object from the map has the url and prefix on it, from line 74

going to rename these and then merge